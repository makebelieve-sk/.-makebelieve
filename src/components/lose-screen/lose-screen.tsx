import React from 'react';
import { Link } from "react-router-dom";

import '../win-screen/win-screen.css';

type LoseScreenType = {
    endTime: boolean,
    reset: () => any
};

export const LoseScreen: React.FC<LoseScreenType> = ({ endTime, reset }) => {
    return (
        <div id="fail-tries" className="fail-tries">
            <section className="result">
                <div className="result__logo"><img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" /></div>
                {!endTime ? (
                    <>
                        <h2 className="result__title">Какая жалость!</h2>
                        <p className="result__total result__total--fail">У вас закончились все попытки. Ничего, повезёт в следующий раз!</p>
                    </>
                ) : (
                    <>
                        <h2 className="result__title">Увы и ах!</h2>
                        <p className="result__total result__total--fail">Время вышло! Вы не успели отгадать все мелодии</p>
                    </>
                )}
                <Link to="/" className="result__replay" onClick={reset}>Попробовать ещё раз</Link>
            </section>
        </div>
    )
};