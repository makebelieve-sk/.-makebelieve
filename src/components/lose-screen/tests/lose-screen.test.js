import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Router } from 'react-router-dom';

import { LoseScreen } from '../lose-screen';
import history from '../../../history';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2) => shallow(
    <LoseScreen endTime={prop1} reset={prop2} />
);

const mock = {
    endTime: false,
    reset: jest.fn()
};

describe(`LoseScreen component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.endTime, mock.reset);
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.fail-tries');
        expect(wrapper.length).toBe(1);
    });

    it(`LoseScreen correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(
                <Router history={history}>
                    <LoseScreen 
                        endTime={mock.endTime}
                        reset={mock.reset}
                    />
                </Router>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});