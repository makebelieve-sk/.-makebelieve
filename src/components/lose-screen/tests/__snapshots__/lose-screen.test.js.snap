// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoseScreen component LoseScreen correctly renders after relaunch 1`] = `
<div
  className="fail-tries"
  id="fail-tries"
>
  <section
    className="result"
  >
    <div
      className="result__logo"
    >
      <img
        alt="Угадай мелодию"
        height="83"
        src="img/melody-logo.png"
        width="186"
      />
    </div>
    <h2
      className="result__title"
    >
      Какая жалость!
    </h2>
    <p
      className="result__total result__total--fail"
    >
      У вас закончились все попытки. Ничего, повезёт в следующий раз!
    </p>
    <a
      className="result__replay"
      href="/"
      onClick={[Function]}
    >
      Попробовать ещё раз
    </a>
  </section>
</div>
`;
